api/v0api/gateway.go

```diff
@@ -65,8 +65,10 @@ type Gateway interface {
 	StateGetAllocationForPendingDeal(ctx context.Context, dealId abi.DealID, tsk types.TipSetKey) (*verifregtypes.Allocation, error)
 	StateGetAllocation(ctx context.Context, clientAddr address.Address, allocationId verifregtypes.AllocationId, tsk types.TipSetKey) (*verifregtypes.Allocation, error)
 	StateGetAllocations(ctx context.Context, clientAddr address.Address, tsk types.TipSetKey) (map[verifregtypes.AllocationId]verifregtypes.Allocation, error)
+	StateGetAllAllocations(ctx context.Context, tsk types.TipSetKey) (map[verifregtypes.AllocationId]verifregtypes.Allocation, error)
 	StateGetClaim(ctx context.Context, providerAddr address.Address, claimId verifregtypes.ClaimId, tsk types.TipSetKey) (*verifregtypes.Claim, error)
 	StateGetClaims(ctx context.Context, providerAddr address.Address, tsk types.TipSetKey) (map[verifregtypes.ClaimId]verifregtypes.Claim, error)
+	StateGetAllClaims(ctx context.Context, tsk types.TipSetKey) (map[verifregtypes.ClaimId]verifregtypes.Claim, error)
```

Comment:
These new endpoints return too much data at once to safely add to the gateway (requires too much buffering). Do you _need_ these on an API gateway? If so, we need to make them stream their results back via a channel.

Reply:
I would expect these to end up being accessed in the same way as statemarketdeals is today - e.g. https://lotus.filecoin.io/lotus/developers/glif-nodes/

Reply:
I agree. you are right -  I mixed glifs customized  end point  for statemarketdeal vs their gateway endpoint in my original issue. @LexLuthr could you please remove gateways? Thanks

Reply:
I have removed the gateway APIs.